src/main/java/com/appdynamics/extensions/azure/customnamespace/azureMonitorExtsCommons/AzureServiceCollector.java

```diff
@@ -90,24 +91,28 @@ public AzureServiceCollector(Builder builder) {
                     executorService.submit("AzureServiceCollector", resourceGroupExecutorTask);
                     futureTasks.add(resourceGroupExecutorTask);
                 } catch (Exception e) {
-                    LOGGER.error("Error while collecting metrics for resourceGroup {}", resourceGroup.name(), e);
+                    LOGGER.error("{} - Error while building resource group task resource group ||{}||", loggingPrefix, resourceGroup.name(), e);
                 }
             }
         } catch (Exception e) {
-            LOGGER.error("Exception while querying for a resourceGroup of account {} and service{}", account.getDisplayName(), service.getServiceName(), e);
-        } finally {
-            return futureTasks;
+            LOGGER.error("{} - Exception in buildFutureTasks()", loggingPrefix, e);
         }
+        return futureTasks;
+        
     }
 
     private List<ResourceGroup> filteredResourceGroups(List<ResourceGroup> resourceGroups, List<String> confResourceGroups) {
+        LOGGER.debug("{} - Filtering available resource groups for configured resource groups ||{}||", loggingPrefix, confResourceGroups);
         List<ResourceGroup> filteredResourceGroup = Lists.newArrayList();
+        List<String> avaialbleResourceGroupNames = Lists.newArrayList();
+        List<String> filteredResourceGroupNames = Lists.newArrayList();
         for (ResourceGroup resourceGroup : resourceGroups) {
+            avaialbleResourceGroupNames.add(resourceGroup.name());
             if (CommonUtilities.checkStringPatternMatch(resourceGroup.name(), confResourceGroups))
+                filteredResourceGroupNames.add(resourceGroup.name());
                 filteredResourceGroup.add(resourceGroup);
```

Comment:
`filteredResourceGroup` will not have the `checkStringPatternMatch` check. It should be as below.
```
            if (CommonUtilities.checkStringPatternMatch(resourceGroup.name(), confResourceGroups)) {
                filteredResourceGroupNames.add(resourceGroup.name());
                filteredResourceGroup.add(resourceGroup);
            }
```

Reply:
Good catch!  Thank you!